libs/infinity_emb/infinity_emb/log_handler.py

```diff
@@ -1,21 +1,273 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2023-now michaelfeil
 
+import json
 import logging
-import sys
+import re
+from contextvars import ContextVar
 from enum import Enum
-from typing import Any
+from functools import wraps
+from time import perf_counter, time
+from traceback import format_exception
+from uuid import uuid4
 
-logging.getLogger().handlers.clear()
+from fastapi import Request, HTTPException
+from opentelemetry.instrumentation.logging import LoggingInstrumentor
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.responses import StreamingResponse
 
-handlers: list[Any] = []
-try:
-    from rich.console import Console
-    from rich.logging import RichHandler
 
-    handlers.append(RichHandler(console=Console(stderr=True), show_time=False))
-except ImportError:
-    handlers.append(logging.StreamHandler(sys.stderr))
+def sla_breached(duration):
+    return duration > 3000
+
+
+class StructuredLogging:
+    request_time = ContextVar("request_time", default=perf_counter())
+    account_id = ContextVar("account_id", default="null")
+    request_id = ContextVar("request_id", default="No request ID")
+    trace_parent = ContextVar("trace_parent", default="-")
+    controller = ContextVar("controller", default=["-"])  # needed for implementation
+
+    _config_dict = {
+        "ts": "%(timestamp_ms)d",
+        "type": "app",
+        "svc": "freddy-infinity",
+        "lvl": "%(levelname)s",
+        "act": "%(controller)s",
+        "a_id": "%(account_id)s",
+        "r_id": "%(request_id)s",
+        "p": "freddy-fs",
+        "tp": "%(trace_parent)s",
+        "thread_id": "%(thread)s",
```

Comment:
We use td.
https://confluence.freshworks.com/display/FRES/Freddy+Log+and+Metrics

Reply:
Done